### ISSUE TITLE
Error for "config file not found" is not super helpful

### ISSUE THREAD

**Issue by dkmiller** [Azure/shrike#66]
This keeps causing user confusion.

See, e.g., https://stackoverflow.microsoft.com/questions/260799 . Currently, the warning isn't super helpful:

> Workspace is not configured. Please update in your configuration file.

**Comment by haozhestat** [comment]
@dkmiller and @fuhuifang : I have found the root cause for which the warning messages "Configuration file does not exist" doesn't  show up in the [logs](https://o365exchange.visualstudio.com/O365%20Core/_build/results?buildId=5460638&view=logs&j=ae69c9bc-6e0c-5b47-c282-32c083076bb0&t=b30ea2c6-570f-5d29-5c17-718c23025219).

[Pull Request 574498: Update error msg when workspace not configured](https://dev.azure.com/msdata/Vienna/_git/aml-ds/pullrequest/574498?path=%2Faml-build-tooling%2Faml_build_tooling%2Fcore%2Fconfiguration.py)

In the script `configuration.py`, the log level of "Configuration file does not exist" is "info". However, the basicConfig has not been set up yet there, so the logging can not been properly initialized in the `configuration.py`.

**Comment by fuhuifang** [comment]
I

> @dkmiller and @fuhuifang : I have found the root cause for which the warning messages "Configuration file does not exist" doesn't show up in the [logs](https://o365exchange.visualstudio.com/O365%20Core/_build/results?buildId=5460638&view=logs&j=ae69c9bc-6e0c-5b47-c282-32c083076bb0&t=b30ea2c6-570f-5d29-5c17-718c23025219).
> 
> [Pull Request 574498: Update error msg when workspace not configured](https://dev.azure.com/msdata/Vienna/_git/aml-ds/pullrequest/574498?path=%2Faml-build-tooling%2Faml_build_tooling%2Fcore%2Fconfiguration.py)
> 
> In the script `configuration.py`, the log level of "Configuration file does not exist" is "info". However, the basicConfig has not been set up yet there, so the logging can not been properly initialized in the `configuration.py`.

Thanks for catching this, Haozhe. I'll create a follow-up pr.

**Comment by fuhuifang** [comment]
@haozhestat 
I did not come up with a good solution to fix this issue, not sure how to initialize `logging.basicConfig` first as we need the config from `configuration.py`. I tried calling `logging.basicConfig()` before running `config = load_configuration()` in `def run()` of `command_line.py` ([link](https://github.com/Azure/shrike/blob/6663a6fa4e18b42135477ef8bd1df32e24a67592/shrike/build/core/command_line.py#L301)), but it didn't help.
Some mitigation: https://github.com/Azure/shrike/pull/68

**Comment by dkmiller** [comment]
@fuhuifang , I think we don't have to log the warning. We can just print it, potentially with additional whitespace to make sure it stands out, e.g.

```
WARNING: the configuration file path provided (`path/to/config.yml`) does not exist inside ((working dir)),
so both preparation and registration will fail.
```

**Comment by haozhestat** [comment]
@fuhuifang and @dkmiller : 

Currently, our practice is that:
- If the path of configuration file provided in the command line doesn't exist, we use the default path of the configuration file;
- If neither of the user provided path and the default path exists, we use the default values of the configuration parameters, which will cause the workspace-not-found error as the default value of workspace parameter is None.

To mitigate this issue, could we just raise a "FileNotFound" error if the path of configuration file provided in the command line doesn't exist

**Comment by dkmiller** [comment]
@haozhestat I agree. If the user explicitly provides a configuration file path, we should raise a "file not found" error if that does not exist.

Same if the don't, and the default configuration file path doesn't exist. Since we _need_ a workspace to be configured, and that can only happen if the user provides a pointer to configuration file.

tl;dr &mdash; if path to config file doesn't exist, throw explicit exception (looks like we're all aligned).

**Comment by haozhestat** [comment]
@fuhuifang and @dkmiller :

Let me summarize the changes we will make in the shrike v1.0.1. Please let me know whether they sound good to you

- If the user provides the configuration file path in the command line:
   - If the user provided path exists, we use it and merge it with the cli arguments and the default configuration
   - If the user provided path doesn't exist, explicitly raise a "file not found" error
- If the user doesn't provide the configuration file path in the command line, we will check whether the default path of configuration file (i.e., `aml-build-configuration.yml`) exists.
   - If the default path of configuration file exists, we use it and merge it with tthe cli arguments and he default configuration;
   - If the default path of configuration file doesn't exist, we will then check whether the `workspaces` is provided in the command line.
      - If the `workspaces` is provided in the command line, it's great. We will merge the cli arguments with the default configuration.
      - If the `workspaces` is provided in the command line, raise an error about "workspace not specified".

**Comment by dkmiller** [comment]
@haozhestat &mdash; users can't provide workspaces via the command line unless I'm missing something.

How to pass in a _list_?

**Comment by haozhestat** [comment]
@dkmiller : You are right! Below is the updated logic:

- If the user provides a configuration file path in the command line:
   - If the user provided path exists, it is great! We merge its content with the cli arguments and the default configuration
   - If the user provided path doesn't exist, we explicitly raise a "file not found" error
- If the user doesn't provide the configuration file path in the command line, we will check whether the default path of configuration file (i.e., `aml-build-configuration.yml`) exists.
   - If the default path of configuration file exists, it is great! We use it and merge it with the cli arguments and the default configuration;
   - If the default path of configuration file doesn't exist, we will raise an error about `No configuration file dectected + no workspace specified` and explicitly ask the user to provide a configuration file.
